app/Http/Controllers/BlogController.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Blog;
+use Illuminate\Http\Request;
+use Illuminate\Support\Str;
+
+class BlogController extends Controller
+{
+    public function index()
+    {
+        $blogs = Blog::all();
+        return view('blog.index', compact('blogs'));
+    }
+
+    public function create()
+    {
+        return view('blog.create');
+    }
+
+    public function store(Request $request)
+    {
+        $request->validate([
+            'title' => 'required|string|max:255',
+            'content' => 'required|string',
+            'slug' => 'nullable|string|unique:blogs,slug|max:255',
+        ]);
+
+        $slug = $request->slug ? $request->slug : Str::slug($request->title);
+        $slug = Blog::where('slug', $slug)->exists() ? $slug . '-' . Str::random(5) : $slug;
+
+        Blog::create([
+            'title' => $request->get("title"),
+            'content' => $request->get("content"),
+            'slug' => $slug,
+        ]);
+
+        return redirect()->route('blog.index')->with('success', 'Blog created successfully!');
+    }
+
+    public function show($slug)
+    {
+        $viewPath = "blog.blogs.$slug";
```

Comment:
Nice how you combined the existing 'static' blog posts with the ones that are in the database. There might be an issue when the user creates a blog post with the same slug as one of the existing ones though.